src/main/java/com/epam/aidial/core/controller/ControllerSelector.java

```
@@ -328,4 +310,18 @@ private Matcher match(Pattern pattern, String path) {
         Matcher matcher = pattern.matcher(path);
         return matcher.find() ? matcher : null;
     }
+
+    private String resourcePath(String url) {
+        String prefix = "/v1/";
+
+        if (!url.startsWith(prefix)) {
```

Nit. It's unnecessary to check the condition. Regexp matcher doesn't pass a wrong url.

---

Yes, but it is more safe approach of doing so, plus it is more obvious for the reader.

---

it's hard to maintain in future, when you have v2, v3 or so. Obviously I don't see a real value of checking this

---

The future is not yet to come. v11 won't work as well.

---

At least it will fail, when you will add a new version. But not silently fail.